### PR TITLE
Zenler new actions and sources

### PR DESCRIPTION
**Pull request by jcortes**
# Actions
- [x] Create User | Creates a user. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#d350a2f6-bc29-b363-6724-681535bc03e0)
- [x] Enroll User | Enrolls a user to a course. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#faaadc39-5702-bb2c-640f-c76a47c26f81)
- [x] Unenroll User | Unenrolls a user from a course. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#faaadc39-5702-bb2c-640f-c76a47c26f82)
- [x] Register Live Class | Registers a live class. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#1966fea2-8274-49bd-f96d-54c215f9d301)
- [x] Register Live Webinar | Registers a live webinar. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#1966fea2-8274-49bd-f96d-54c215f9d302)
- [x] Subscribe Funnel | Subscribes to a funnel. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#subscribe_to_funnel)
- [ ] ~Register Live Interactive Webinar | Registers a live interactive webinar~ I didn't find an endpoint for this component

# Sources
- [x] New User | Emit new event when a user is created. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#9417d5e4-9951-0933-f562-593e946b52a6)
- [x] Course Complete | Emit new event when a course is completed. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#e0160d58-f0c5-8264-7ee2-fb991cd33e1b)
- [x] New Funnel Subscription | Emit new event when a funnel is created. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#0052da8d-ca30-b23b-48b3-5cbdce72547e)
- [x] New Live Class Registration | Emit new event when a new live class is registered. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#1966fea2-8274-49bd-f96d-54c215f9d303). ***I couldn't test this component because it requires and upgraded plan in order to register these resources***
- [x] New Live Interactive Webinar Registration | Emit new event when a new live interactive webinar is registered. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#1966fea2-8274-49bd-f96d-54c215f9d304) ***I couldn't test this component because it requires and upgraded plan in order to register these resources***
- [x] Lesson Complete | Emit new event when a lesson is completed. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#e0160d58-f0c5-8264-7ee2-fb991cd33e1b)
- [ ] ~New Sale | Emit new event when a sale is created. [See the docs here](https://www.newzenler.com/api/documentation/public/api-doc.html#5b964b29-b972-ec8e-3e65-df2ff33a6ad8)~ There is no implementation due to integration dependency with money transaction platforms. E.g. stripe

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4312"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/zenler/actions/create-user/create-user.mjs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    commission: {
+      type: "integer",
+      label: "Commission",
+      description: "Commission of the user",
+    },
+    roles: {
+      type: "string[]",
+      label: "Roles",
+      description: "Roles of the user",
+      options: [
```
Comment (vellames) [inline components/zenler/actions/create-user/create-user.mjs:40]: Could you please move these options to an options file?

File: components/zenler/actions/enroll-user/enroll-user.mjs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    const response = await this.zenler.enrollUser({
+      userId,
+      data: {
+        course_id: courseId,
+      },
+    });
+
+    if (typeof(response) === "string") {
+      console.log(response);
+      throw new Error("Response error");
```
Comment (vellames) [inline components/zenler/actions/enroll-user/enroll-user.mjs:39]: Is it possible to specify the error in the `throw` message?

File: components/zenler/actions/register-live-class/register-live-class.mjs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      data: {
+        name,
+        email,
+        last_name: lastName,
+      },
+    });
+
+    if (typeof(response) === "string") {
+      console.log(response);
+      throw new Error("Response error");
```
Comment (vellames) [inline components/zenler/actions/register-live-class/register-live-class.mjs:53]: Is it possible to specify the error in the `throw` message?

File: components/zenler/actions/register-live-webinar/register-live-webinar.mjs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    const response = await this.zenler.registerLiveWebinar({
+      webinarId,
+      data: {
+        email,
+      },
+    });
+
+    if (typeof(response) === "string") {
+      console.log(response);
+      throw new Error("Response error");
```
Comment (vellames) [inline components/zenler/actions/register-live-webinar/register-live-webinar.mjs:38]: Is it possible to specify the error in the `throw` message?

File: components/zenler/actions/subscribe-funnel/subscribe-funnel.mjs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        email,
+        last_name: lastName,
+        address,
+        city,
+      },
+    });
+
+    if (typeof(response) === "string") {
+      console.log(response);
+      throw new Error("Response error");
```
Comment (vellames) [inline components/zenler/actions/subscribe-funnel/subscribe-funnel.mjs:68]: Is it possible to specify the error in the `throw` message?

File: components/zenler/actions/unenroll-user/unenroll-user.mjs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    const response = await this.zenler.unenrollUser({
+      userId,
+      data: {
+        course_id: courseId,
+      },
+    });
+
+    if (typeof(response) === "string") {
+      console.log(response);
+      throw new Error("Response error");
```
Comment (vellames) [inline components/zenler/actions/unenroll-user/unenroll-user.mjs:39]: Is it possible to specify the error in the `throw` message?

File: components/zenler/package.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "@pipedream/zenler",
+  "version": "0.0.1",
+  "description": "Pipedream Zenler Components",
+  "main": "zenler.app.mjs",
+  "keywords": [
+		"pipedream",
```
Comment (vellames) [inline components/zenler/package.json:7]: Looks like there is an extra tab here

**Review by vellames — CHANGES_REQUESTED** [review]
Hi there. I just left some suggestions

File: components/zenler/actions/enroll-user/enroll-user.mjs (hunk 2 above)
Comment (jcortes) [reply components/zenler/actions/enroll-user/enroll-user.mjs:39]: Actually when this happens the response comes in with HTML format, and it doesn't say much. It just says something like "Something went wrong" that's why I've just put `Response error`

File: components/zenler/actions/register-live-webinar/register-live-webinar.mjs (hunk 4 above)
Comment (jcortes) [reply components/zenler/actions/register-live-webinar/register-live-webinar.mjs:38]: See [here](https://github.com/PipedreamHQ/pipedream/pull/4312#discussion_r981511609)

File: components/zenler/actions/unenroll-user/unenroll-user.mjs (hunk 6 above)
Comment (jcortes) [reply components/zenler/actions/unenroll-user/unenroll-user.mjs:39]: See [here](https://github.com/PipedreamHQ/pipedream/pull/4312#discussion_r981511609)

File: components/zenler/package.json (hunk 7 above)
Comment (jcortes) [reply components/zenler/package.json:7]: Actually there is no extra tab as far as I can tell

File: components/zenler/actions/enroll-user/enroll-user.mjs (hunk 2 above)
Comment (vellames) [reply components/zenler/actions/enroll-user/enroll-user.mjs:39]: Got it! tks!

**Review by vellames — APPROVED** [review]
lgtm!

**Comment by DilanAthukorala** [comment]
@jcortes Observing following issues;

1. for enroll_user , register_live_class , register_live_webinar , subscribe_funnel , unenroll_user
 select dropdown props are disabled before refreshing the page each time after the action is added. 
image
image
image
image
image



2. for new [New Lesson Complete] and [New Course Complete] sources no events are emited
image

**Comment by jcortes** [comment]
Hi @DilanAthukorala seems like the first issues has to do more with an issue in the UI not related to the components however I couldn't reproduce it myself.

About the second issue regarding `New Course Completed` source you need at least to create a course and enroll at least one student to that course to see events happening.
Also regarding `New Lesson Complete` you need a course and a few lessons in that course in order to get events. For this issue I made some changes so you can test them again plaase!

**Review by vellames — APPROVED** [review]

**Comment by DilanAthukorala** [comment]
@jcortes Thanks, I think there was a delay in emitting events I received the old events now. 

Verification completed and ready for release.

**Pull request merged**